src/config.ts

```diff
@@ -0,0 +1,109 @@
+    return Reflect.set(target, key, value)
+  }
+})
+
+const serializeConfig = <T extends PartialTWConfig>(config: T) =>
+  JSON.stringify(
+    configMerger({ plugins: () => [] }, config),
+    (_, v) => typeof v === 'function' ? `() => (${JSON.stringify(v())})` : v).replace(/"(\(\) => \(.*\))"/g, (_, substr) => substr.replace(/\\"/g, '"')
+  )
+
+/**
+ * Loads all possible Tailwind CSS configuration for a Nuxt project.
+ *
+ * @param moduleOptions configuration for the module
+ * @param nuxt nuxt app
+ * 
+ * @returns template with all configs handled
+ */
+export default async function loadTwConfig(moduleOptions: ModuleOptions, nuxt = useNuxt()) {
+  const [configPaths, contentPaths] = await resolveModulePaths(moduleOptions.configPath, nuxt)
+  const configProxies: [string, PartialTWConfig][] = configPaths.map((configPath) => [configPath, {}])
+
+  const tailwindConfig = await Promise.all((
+    configPaths.map(async (configPath, idx, paths) => {
+      let _tailwindConfig: PartialTWConfig | undefined
+
+      try {
+        _tailwindConfig = loadConfig(configPath)
+      } catch (e) {
+        logger.warn(`Failed to load Tailwind config at: \`./${relative(nuxt.options.rootDir, configPath)}\``, e)
+      }
+
+      // Transform purge option from Array to object with { content }
+      if (_tailwindConfig && !_tailwindConfig.content) {
+        configProxies[idx][1].content = _tailwindConfig.purge
+      }
+
+      await nuxt.callHook('tailwindcss:loadConfig', _tailwindConfig && new Proxy(_tailwindConfig, makeHandler(configProxies[idx][1])), configPath, idx, paths)
+      return _tailwindConfig || {}
+    }))
+  ).then((configs) => configs.reduce(
+    (prev, curr) => configMerger(curr, prev),
+    // internal default tailwind config
+    configMerger(moduleOptions.config, { content: contentPaths })
+  )) as TWConfig
+
+  // Allow extending tailwindcss config by other modules
+  const configProxy: PartialTWConfig = {}
+  await nuxt.callHook('tailwindcss:config', new Proxy(tailwindConfig, makeHandler(configProxy)))
+
+  const mergerTemplate = addTemplate({ src: createResolver(import.meta.url).resolve('./runtime/merger.mjs'), write: true })
+
+  return {
+    template: addTemplate({
+      filename: 'tailwind.config.cjs',
+      write: true,
+      getContents: () => {
+        return [
+          `const configMerger = require(${JSON.stringify(mergerTemplate.dst)});`,
+          configPaths.map((p, idx) => `const cfg${idx} = require(${JSON.stringify(/[/\\]node_modules[/\\]/.test(p) ? p : './' + relative(nuxt.options.buildDir, p))});`).join('\n'),
+          `\nconst inlineConfig = ${serializeConfig(moduleOptions.config as PartialTWConfig)};\n`,
+          `const config = [`,
+          configProxies.map(([_, p], idx) => `  configMerger(${serializeConfig(p)}, cfg${idx})`).join(', \n'),
+          `].reduce((prev, curr) => configMerger(curr, prev), configMerger(inlineConfig, { content: ${JSON.stringify(contentPaths)} }));\n`,
```

Comment:
The thing is, this works and it's great, but I feel there are wayyyy too many configMerger calls in this function and while nothing is asynchronous here, the complexity and expense is not negligible. Let's see how this can be improved.

Reply:
This affects tailwindcss config loading time, so HMR is still a few ms late and not as-instant.